lib/js/src/rpc/messages/RegisterAppInterface.js

```diff
@@ -0,0 +1,381 @@
+    *                                     that is audible outside of the BT media source.
+    * @return {RegisterAppInterface}
+    */
+    setIsMediaApplication(isMediaApplication) {
+        this.setParameter(RegisterAppInterface.KEY_IS_MEDIA_APPLICATION, isMediaApplication);
+        return this;
+    }
+
+    /**
+    * @return {Boolean}
+    */
+    getIsMediaApplication() {
+        return this.getParameter(RegisterAppInterface.KEY_IS_MEDIA_APPLICATION);
+    }
+
+    /**
+    * @param {Language} languageDesired
+    * @return {RegisterAppInterface}
+    */
+    setLanguageDesired(languageDesired) {
+        this.validateType(Language, languageDesired);
+
+        this.setParameter(RegisterAppInterface.KEY_LANGUAGE_DESIRED, languageDesired);
+        return this;
+    }
+
+    /**
+    * @return {Language}
+    */
+    getLanguageDesired() {
+        return this.getObject(Language, RegisterAppInterface.KEY_LANGUAGE_DESIRED);
+    }
+
+
+    /**
+    * @param {Language} hmiDisplayLanguageDesired
+    * @return {RegisterAppInterface}
+    */
+    setHmiDisplayLanguageDesired(hmiDisplayLanguageDesired) {
+        this.validateType(Language, hmiDisplayLanguageDesired);
+
+        this.setParameter(RegisterAppInterface.KEY_HMI_DISPLAY_LANGUAGE_DESIRED, hmiDisplayLanguageDesired);
+        return this;
+    }
+
+    /**
+    * @return {Language}
+    */
+    getHmiDisplayLanguageDesired() {
+        return this.getObject(Language, RegisterAppInterface.KEY_HMI_DISPLAY_LANGUAGE_DESIRED);
+    }
+
+
+    /**
+    * @param {Array<AppHMIType>} appHMIType
+    * @return {RegisterAppInterface}
+    */
+    setAppHmiType(appHMIType) {
+        //TODO make validate type accept arrays 
+        //this.validateType(AppHMIType, appHMIType);
+
+        this.setParameter(RegisterAppInterface.KEY_APP_HMI_TYPE, appHMIType);
+        return this;
+    }
+
+    /**
+    * @return {Array<AppHMIType>}
+    */
+    getAppHmiType() {
+        return this.getObject(AppHMIType, RegisterAppInterface.KEY_APP_HMI_TYPE);
+    }
+
+    /**
+    * @param {String} hashID the hash ID
+    * @return {RegisterAppInterface}
+    */
+    setHashID(hashID) {
+        //TODO do we need this? this.validateType(String, hashID);
+
+        this.setParameter(RegisterAppInterface.KEY_HASH_ID, hashID);
+        return this;
+    }
+
+    /**
+    * @return {String} the hash ID
+    */
+    getHashID() {
+        return this.getParameter(RegisterAppInterface.KEY_HASH_ID);
+    }
+
+    /**
+     * @param {DeviceInfo} deviceInfo
+     * @return {RegisterAppInterface}
+     */
+    setDeviceInfo(deviceInfo) {
+        this.validateType(DeviceInfo, deviceInfo);
+
+        this.setParameter(RegisterAppInterface.KEY_DEVICE_INFO, deviceInfo);
+        return this;
+    }
+
+    /**
+    * @return {DeviceInfo}
+    */
+    getDeviceInfo() {
+        return this.getParameter(RegisterAppInterface.KEY_DEVICE_INFO);
```

Comment:
As I understand there should be `this.getObject(DeviceInfo, RegisterAppInterface.KEY_DEVICE_INFO);` instead of your `getParameter` is it right?